packages/nimble-components/src/table/specs/table-column-placeholder-hld.md

```diff
@@ -0,0 +1,151 @@
+# Placeholders for table columns HLD
+
+## Problem Statement
+
+In some cases, an application may want to display a placeholder value in the table when a record does not have a value for a given field. The application should be able to customize that placeholder based on their needs.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Nimble issue 1538](https://github.com/ni/nimble/issues/1538)
+-   [Nimble issue 1511](https://github.com/ni/nimble/issues/1511)
+
+## Implementation / Design
+
+### High-level behavior
+
+There are two places where a value is displayed in the table that placeholders need to be considered: (1) in a cell and (2) in a group row. The cell placeholder will be configurable on each column that supports having a placeholder, and it will default to an empty string. The group row placeholder will come from the table's localization provider, and it will not be configurable through a column's API.
+
+Placeholders within a cell will be rendered with nimble's placeholder font, which is currently 60% opacity of nimble's body font.
+
+Placeholders within a group row will have no special visual treatment.
+
+Below is an example of what placeholders will look like in the table. In this example, the table is grouped by the "Quote" column, which has been configured to have a placeholder value of "None".
+
+![Placeholder text example](./spec-images/PlacholderText.png)
+
+### Column-specific decisions
+
+The exact behavior of placeholders in each existing table column is described below.
+
+#### Text column
+
+| Special-cased field values | Cell display                                                 | Group row display |
+| -------------------------- | ------------------------------------------------------------ | ----------------- |
+| `undefined`                | column placeholder, or empty if no placeholder is configured | `"No value"`      |
+| `null`                     | column placeholder, or empty if no placeholder is configured | `"No value"`      |
+| `''`                       | \<empty cell>                                                | `"Empty"`         |
+
+#### Anchor column
+
+| Special-cased field values                          | Cell display                                                       | Group row display |
+| --------------------------------------------------- | ------------------------------------------------------------------ | ----------------- |
+| Both label and href are `undefined` or `null`       | column placeholder, or empty if no placeholder is configured       | `"No alias"`      |
```

Comment:
The tricky thing is that (at least today) the grouping code can only look at one field in the record. Therefore, we can't make a distinction between the `undefined` label + `undefined` href case and the `undefined` label + defined href case when grouping. All of those values will get grouped together since we are grouping on the label.

Reply:
In the other comment I make the argument that having an undefined or null label with an href is not an application recommendation. If that actually makes sense then a well behaving app shouldn't be leveraging the case of (null/undefined label + href) and (null/undefined label + null/undefined href) at the same time. If that's convincing then we could just have the "No value" label and skip the "No Alias" label?

Reply:
That makes sense to me, and it certainly simplifies things.

@fredvisser, any objection to getting rid of the "No alias" label? Instead, we would show "No value" any time the label is `undefined`/`null`, and we will recommend against ever having `undefined`, `null`, or empty labels associated with a defined href value.

Reply:
@fredvisser, wanted to check that you had seen this conversation and see if you had any opinions.

Reply:
No objection to getting rid of the "No alias". Client apps should clean up their data, such that they don't have a URL with an empty label.